bitcointranscripts/main.py

```diff
@@ -0,0 +1,143 @@
+import asyncio
+import os
+import re
+import zipfile
+from datetime import datetime
+import traceback
+
+import requests
+import yaml
+from dotenv import load_dotenv
+from loguru import logger
+
+from common.elasticsearch_utils import upsert_document
+
+load_dotenv()
+
+folder_name = "bitcointranscripts-master"
+index_name = os.getenv('INDEX')
+
+
+def download_repo():
+    URL = "https://github.com/bitcointranscripts/bitcointranscripts/archive/refs/heads/master.zip"
+    data_dir = os.getenv('DATA_DIR')
+    dir_path = os.path.join(data_dir, "bitcointranscripts")
+    os.makedirs(dir_path, exist_ok=True)
+
+    if os.path.exists(os.path.join(dir_path, folder_name)):
+        logger.info(f"Repo already downloaded at path: {dir_path}")
+        return
+
+    logger.info(f"Downloading repo at path: {dir_path}")
+    file_path = os.path.join(dir_path, "master.zip")
+
+    # Download the file
+    response = requests.get(URL)
+    with open(file_path, 'wb') as file:
+        file.write(response.content)
+
+    logger.info(f"Downloaded {URL} to {file_path}")
+
+    # Unzip
+    with zipfile.ZipFile(file_path, 'r') as zip_ref:
+        zip_ref.extractall(dir_path)
+
+    logger.info(f"Unzipped {file_path} to {dir_path}")
+
+
+def parse_posts(directory):
+    documents = []
+    root_depth = directory.rstrip(os.sep).count(os.sep)
+    for root, dirs, files in os.walk(directory):
+        current_depth = root.rstrip(os.sep).count(os.sep)
+        if current_depth <= root_depth:
+            continue
+        for file in files:
+            if file.endswith('.md') and not file.startswith('_') and not re.search(r'\.([a-z][a-z])\.md$', file):
+                file_path = os.path.join(root, file)
+                document = parse_post(file_path)
+                documents.append(document)
+    return documents
+
+
+def parse_post(file_path):
+    with open(file_path, 'r', encoding='utf-8') as file:
+        content = file.read()
+
+    # Remove content between {% %}
+    content = re.sub(r'{%.*%}', '', content, flags=re.MULTILINE)
+    front_matter, body = parse_markdown(content)
+
+    # Extract metadata from front matter using yaml
+    metadata = yaml.safe_load(front_matter)
+
+    custom_id = file_path.replace('.md', '').replace(os.path.join(os.getenv('DATA_DIR'), "bitcointranscripts", folder_name), '')
```

Comment:
you are already assigning `os.path.join(os.getenv('DATA_DIR'), "bitcointranscripts", folder_name` in a variable in the `main` method. Does it make sense to define it on the top and reuse it here as well as in the `download_repo` method? This will also improve readability.